content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,573 @@
+
+To learn more about label encoding, [click](/engineering-education/data-preprocessing-in-r/).
+ 
+```py
+def species_cat(df):
+ if df["species"] == "Iris-virginica":
+   return 0
+ elif df["species"] == "Iris-versicolor":
+   return 1
+ else:
+   return 2
+ 
+df["species"] = df.apply(species_cat, axis=1)
+```
+
+```py
+print(df.sample(5))
+```
+
+![dfsample5](/engineering-education/statistical-hypothesis-testing-python-implementation/dfsample5.png)
+
+```py
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+print(df_new.head())
+```
+
+![newhead](/engineering-education/statistical-hypothesis-testing-python-implementation/df_new.png)
+ 
+Let's make use of the `chic2_contigency` library that we had imported into the project.
+
+```python
+stat, p, dof, expected= chi2_contingency(df_new) 
+print("The degree of freedom is: ", dof)
+```
+
+Output:
+
+```bash
+The degree of freedom is: 149
+```
+
+The three outputs in test statistics will include degree of freedom, p-value, and expected values.
+ 
+[Test statistic](https://online.stat.psu.edu/statprogram/reviews/statistical-concepts/chi-square-tests) is a characterized feature of significance in the Chi-square test. Its helps in determining how uncommon the result might be provided that null hypothesis is assumed to be true.
+ 
+The number of features or variables that we have access to vary depending on the [degree of freedom](https://byjus.com/maths/chi-square-test/). This is why the probability of confidence is set to 95%.
+
+```py
+prob = 0.95
+ 
+alpha = 1.0 - prob
+ 
+print('The alpha/significance level=%.3f' % alpha)
+print('The P-value is =%.2f' % p)
+ 
+if p <= alpha:
+ print('Reject the Null Hypothesis ( reject H0)')
+else:
+ print('Accept the Null Hypothesis (fail to reject H0)')
+```
+
+Output:
+
+```bash
+alpha/significance level=0.050
+P-value = 0.10
+Accept the Null Hypothesis (fail to reject H0)
+```
+
+In the results above, the `p-value` is greater than the `alpha value` set at 0.05. We will accept the null hypothesis and reject the alternate hypothesis. This means the petal width and species of flower are not dependent, i.e. there is no relationship between them.
+ 
+### T-Test with Python implementation
+The T-test is utilized when we plan to evaluate the discrepancy between the means of two groups of samples. Unlike the Chi-Square Test, the T-test is used on continuous variables.
+ 
+T-tests has three basic types
+ 
+1. **One sample t-test:** It checks whether the population has a different mean from the sample.
+The mathematical expression is:
+
+[one-sample-t-test](/engineering-education/statistical-hypothesis-testing-python-implementation/one-sample-t-test.PNG)
+ 
+where,
+the observed mean group is a,
+ 
+the assumed population means is B,
+ 
+the standard deviation of the data group is X,
+ 
+the number of observations in the group is N.
```

Comment:
Remove the extra whiteline between the statements. Also, the variables themselves should be in inverted quotes. For example,

"the observed mean group is `a`,"